### PR TITLE
Remote Signing and Slashing Protection

### PR DESCRIPTION
**Pull request by skardas**
# Grant Application Checklist

- [x] The [application-template.md](https://github.com/w3f/Open-Grants-Program/blob/master/applications/application-template.md) has been copied, renamed ( "project_name.md") and updated.
- [x] A BTC or Ethereum address for the payment of the milestones is provided inside the application.  
- [x] The software of the project will be released under the Apache license version 2.0 as specified in the [terms and conditions](https://gist.github.com/Noc2/75bc58e8ce9b5d419ff883b0cf2b8c19).
- [x] The total funding amount of the project is below USD $30k for initial grant applications and $100k for follow-up grants at the time of submission.
- [x] The initial PR contains only one commit (squash if needed before submitting your PR).
- [x] The grant will only be announced once we successfully [delivered](https://github.com/w3f/Grant-Milestone-Delivery) the first milestone.

### CODE REVIEW

**Review by skardas — COMMENTED** [review]
ok

**Review by fruit37 — APPROVED** [review]

**Review by mehmetankr — APPROVED** [review]

**Review by dmitry123 — APPROVED** [review]

**Review by mmagician — CHANGES_REQUESTED** [review]
@skardas Thanks for the application, I think remote signing could be a good addition to the operation of validators. However, I have a few comments:

1. Approving your own PR doesn't help, only authorised users' approvals count
2. We don't fund the design & specification phase - this should comprise part of your initial research into the technology and should be submitted as part of the application itself. Furthermore, having developed a similar mechanism for eth already, it sounds like you should be able to provide a detailed architecture overview already.
3. > Providing a crypto library that supports ECDSA, Ed25519, and sr25519 signature algorithms

These libraries already exist. Could you outline what exactly is your contribution here?

4. > To add slashing prevention we also need to pass payload details into backend. 

How is this going to prevent slashing? Could you outline how the backend is going to process/store this information in order to prevent slashing?

5. > However, it should also be implemented in Polkadot codebase directly (i.e., not in Substrate).

Please elaborate on this point, as it's not entirely clear what you mean here.

6. > Providing a remote server

It's alright that you run a server yourself, but what we require for the Grants Program is to have the code open sourced, so that users can, if they choose, run such a server themselves. Are you fine with doing this?

To summarise, I think the grant could potentially make an impact on how validators process tx's, but please review the above points before we consider the application further.

**Comment by mmagician** [comment]
Closing in favour of https://github.com/w3f/Open-Grants-Program/pull/237
